### PR TITLE
Core revamp part 3

### PR DESCRIPTION
**Pull request by prasannavl**
## Description

- Extract form.rs into tide-forms
- Extract querystring.rs into tide-querystring
- Refactor `tide` lib.rs
- Move shared internals into `tide_core::internal`

## Motivation and Context

https://github.com/rustasync/tide/pull/252, #162 

## How Has This Been Tested?

All tests passed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactor

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by fairingrey** [comment]
Looks great! Although I will remark that we should probably rename all occurrences of `Data` to `State` on the numerous `ContextExt` traits impl'd by each of these now sub-crates.

Unrelated, given, but just something I've noticed that we haven't fully accounted for.

**Comment by prasannavl** [comment]
@fairingrey - I think a separate PR with full sweep of it should be readily merge-able.  (Could you open an issue? It'll make a good first issue too)

File: tide-core/src/error.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ impl std::fmt::Display for StringError {
 #[macro_export]
 macro_rules! err_fmt {
     {$($t:tt)*} => {
-        crate::error::StringError(format!($($t)*))
+        $crate::error::StringError(format!($($t)*))
```
Comment (yoshuawuyts) [inline tide-core/src/error.rs]: Was this intentional?

**Comment by fairingrey** [comment]
With 4 approvals, I'll go ahead and merge this :+1:

File: tide-core/src/error.rs (hunk 1 above)
Comment (prasannavl) [reply tide-core/src/error.rs]: Hi @yoshuawuyts - yes! This is needed since `err_fmt` is in tide_core and also re-exported by `tide`.

**Pull request merged**